### PR TITLE
Improve network status check

### LINKED ISSUE #933: Improve network status check
NetworkUtils.java considers the state 'connecting' on mobile 'OK'

1. The network state check should be more restrictive, e.g. a download operation should pause also if being in state 'connecting', as the respecting network login process may fail.
2. The state info from the service is not checked against null and for 'connected' always.
3. The respective tests have no annotation and fail if annotated as test.

Please review the proposal and comments in the fork branch 
[/chris1b/kiwix-android/tree/feature/improve_network_status_check](https://github.com/chris1b/kiwix-android/tree/feature/improve_network_status_check)

I will cleanup/simplify the proposed code upon your input.

### PR DESCRIPTION
**Pull request by chris1b**
proposal to fix #933

Changes: 
add checks for null and status 'connected'
add comments to be reviewed
enable and fix existing tests

TODO upon review statements: 
remove comments
probably remove option to check for network 'availability' (opposed to network 'connection')

### CODE REVIEW

File: app/src/main/java/org/kiwix/kiwixmobile/utils/NetworkUtils.java [hunk 1]
```diff
@@ -31,17 +31,28 @@
 
 public class NetworkUtils {
 
+  //TODO verify if NetworkInfo.State#CONNECTING is really sufficient: Should we not check for state 'connected'?
+  /**
```
Comment (mhutti1) [inline app/src/main/java/org/kiwix/kiwixmobile/utils/NetworkUtils.java]: Yeah this sounds right, we should check for connected

File: app/src/main/java/org/kiwix/kiwixmobile/utils/NetworkUtils.java [hunk 2]
```diff
@@ -50,21 +61,31 @@ public static boolean isNetworkAvailable(Context context) {
...
 
+  /**
+   * @see #isNetworkAvailable(Context, boolean)
+   */
+  static boolean isNetworkConnectionOK(NetworkInfo networkInfo, boolean tolerateStateConnecting) {
+    return networkInfo.getState() == NetworkInfo.State.CONNECTED
+      || (tolerateStateConnecting && networkInfo.getState() == NetworkInfo.State.CONNECTING);
+  }
+
+  //TODO method isWiFi should be renamed to isWifiConnected to express the state which is checked
```
Comment (mhutti1) [inline app/src/main/java/org/kiwix/kiwixmobile/utils/NetworkUtils.java]: Yeah we should do this

**Comment by soloturn** [comment]
thanks @chris1b for tackling the piece of kiwix which gives it a bad rating in playstore! i tried yesterday:

1. switch off mobile data
2. connect wifi
3. start download
4. cut wifi
5. connect wifi again

there are a lot of smaller and bigger issues coming up on this way. status messages not disappearing, no restart of download, crash. everything is there.

**Comment by soloturn** [comment]
if you would be able to have also a look into:

* what is deprecated now: 
* https://stackoverflow.com/questions/32547006/connectivitymanager-getnetworkinfoint-deprecated
* https://stackoverflow.com/questions/32242384/android-getallnetworkinfo-is-deprecated-what-is-the-alternative
* where it is used, and how to catch when the connection breaks and restarts
* how to pass this onto the piece doing the download
* displaying the status
 
i know its huge, but maybe it can be broken apart, as you already started.

**Comment by soloturn** [comment]
@iadeelzafar seemed to found one related to here, in @998.

**Comment by kelson42** [comment]
@chris1b Please rebase your branch properly

**Comment by chris1b** [comment]
> 
> 
> @chris1b Please rebase your branch properly

@kelson42 Sorry, I pushed the rebased master now (as I did the feature branch before), but the conflicts remain listed here. I can't see if a travis rescan is missing/can be triggered?

**Comment by mhutti1** [comment]
@chris1b Github is still showing me over 200 files changed.

**Comment by kelson42** [comment]
@chris1b Have you given up?

**Comment by kelson42** [comment]
@chris1b `git rebase master` in your branch should do the job I think.

**Comment by chris1b** [comment]
@kelson42 thanx for your hint: here's what I did today:
1. on local master branch: git pull upstream master
1. on local feature branch: git rebase master using 'accept theirs' in all cases except the 2 files NetworkUtils*.java

git status now shows:
```
On branch feature/improve_network_status_check
Your branch and 'origin/feature/improve_network_status_check' have diverged,
and have 41 and 49 different commits each, respectively.
  (use "git pull" to merge the remote branch into yours)
nothing to commit, working directory clean
```
Is it relevant for the maintainer if I do a merge or a rebase on my two feature branches? (I think: no)

**Pull request closed without merging**